[Bug]: Wrong type of returned value for HuggingFaceInferenceAPIEmbedding::_async_embed_single

### Bug Description

`_async_embed_single` should return `Embedding/List[float]` but it returns `List[np.float32]`
This is because huggingface_hub's `AsyncInferenceClient::feature_extraction` returns `np.array[np.float32]` and the values are not converted to standard float.
https://github.com/run-llama/llama_index/blob/main/llama_index/embeddings/huggingface.py#L238
This is a problem for components that expect a list of regular floats. In particular vector stores may not support numpy float32.

### Version

HEAD

### Steps to Reproduce

Example of failing test
```python
def test_hugging_face():
        embed_model = HuggingFaceInferenceAPIEmbedding(
            model_name="facebook/bart-base", token="xxxxxxxx"
        )
        embeddings = embed_model.get_query_embedding("Hello world")
>       assert type(embeddings[0]) == float
E       AssertionError: assert <class 'numpy.float32'> == <class 'float'>
```

### Relevant Logs/Tracbacks

_No response_

--- a/llama_index/embeddings/huggingface.py
+++ b/llama_index/embeddings/huggingface.py
@@ -238,9 +238,9 @@ class HuggingFaceInferenceAPIEmbedding(HuggingFaceInferenceAPI, BaseEmbedding): 
     async def _async_embed_single(self, text: str) -> Embedding:
         embedding = (await self._async_client.feature_extraction(text)).squeeze(axis=0)
         if len(embedding.shape) == 1:  # Some models pool internally
-            return list(embedding)
+            return embedding.tolist()
         try:
-            return list(self.pooling(embedding))  # type: ignore[misc]
+            return self.pooling(embedding).tolist()  # type: ignore[misc]
         except TypeError as exc:
             raise ValueError(
                 f"Pooling is required for {self.model_name} because it returned"

--- a/tests/embeddings/test_huggingface.py
+++ b/tests/embeddings/test_huggingface.py
@@ -42,7 +42,9 @@ class TestHuggingFaceInferenceAPIEmbeddings:
     def test_embed_query(
         self, hf_inference_api_embedding: HuggingFaceInferenceAPIEmbedding
     ) -> None:
-        raw_single_embedding = np.random.rand(1, 3, 1024)
+        raw_single_embedding = np.random.default_rng().random(
+            (1, 3, 1024), dtype=np.float32
+        )
 
         hf_inference_api_embedding.pooling = Pooling.CLS
         with patch.object(
@@ -53,6 +55,7 @@ class TestHuggingFaceInferenceAPIEmbeddings:
             embedding = hf_inference_api_embedding.get_query_embedding("test")
         assert isinstance(embedding, list)
         assert len(embedding) == 1024
+        assert isinstance(embedding[0], float)
         assert np.all(
             np.array(embedding, dtype=raw_single_embedding.dtype)
             == raw_single_embedding[0, 0]
@@ -68,6 +71,7 @@ class TestHuggingFaceInferenceAPIEmbeddings:
             embedding = hf_inference_api_embedding.get_query_embedding("test")
         assert isinstance(embedding, list)
         assert len(embedding) == 1024
+        assert isinstance(embedding[0], float)
         assert np.all(
             np.array(embedding, dtype=raw_single_embedding.dtype)
             == raw_single_embedding[0].mean(axis=0)

